# IS 362 – Week 5 Assignment
## Your task in this week’s assignment is to answer three questions:
1. What is the northernmost airport in the United States?

In [6]:
import pandas as pd
airports = pd.read_csv("https://raw.githubusercontent.com/tidyverse/nycflights13/main/data-raw/airports.csv")
airports.head()
airports.sort_values("lat", ascending=False).head(8)

faa                                        name        lat         lon  \
417   EEN                     Dillant Hopkins Airport  72.270833   42.898333   
230   BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110   AIN                          Wainwright Airport  70.638056 -159.994722   
708   K03                               Wainwright As  70.613378 -159.860350   
152   ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   
1363  UUK                        Ugnu-Kuparuk Airport  70.330833 -149.597500   
982   NUI                             Nuiqsut Airport  70.210000 -151.005556   
1197  SCC                                   Deadhorse  70.194750 -148.465167   

      alt  tz dst              tzone  
417   149  -5   A                NaN  
230    44  -9   A  America/Anchorage  
110    41  -9   A  America/Anchorage  
708    35  -9   A  America/Anchorage  
152    96  -9   A  America/Anchorage  
1363   67  -9   A  America/Anchorage  
982    38  -9   A  America/Anchorage  
1197   64  -9   A  America/Anchorage

### According to the data collected from the csv file Dillant Hopkins Airport is the northernmost airport. However, upon fact checking it appears the latitude and longitude displayed is incorrect, with Dillant actually have a latitude of 42 and Longitude of -72. In comparison, Wiley Post, the actually northernmost airport, has as latitude of 35 and longitude of -95. The csv file seems to have a discrepancy regarding its inputted data.

2. What is the easternmost airport in the United States?

In [7]:
df = pd.read_csv('https://raw.githubusercontent.com/tidyverse/nycflights13/main/data-raw/airports.csv')
northeast_airports = df[(df['lat'] > 44) & (df['lat'] <= 50) & (df['lon'] >= -75) & (df['lon'] <= -65)]
northeast_airports.sort_values('lat', ascending=False)


faa                                 name        lat        lon   alt  \
1398  WFK  Northern Aroostook Regional Airport  47.285556 -68.312778   988   
259   CAR                         Caribou Muni  46.871500 -68.017917   626   
1101  PQI  Northern Maine Rgnl At Presque Isle  46.688958 -68.044797   534   
624   HUL                         Houlton Intl  46.123083 -67.792056   489   
894   MLT                     Millinocket Muni  45.647836 -68.685561   408   
923   MSS            Massena Intl Richards Fld  44.935833 -74.845547   215   
444   EPM           Eastport Municipal Airport  44.910111 -67.012694    45   
191   BGR                          Bangor Intl  44.807444 -68.828139   192   
1053  PBG                     Plattsburgh Intl  44.650944 -73.468139   234   
933   MVL      Morrisville Stowe State Airport  44.535000 -72.614000   732   
236   BTV                      Burlington Intl  44.471861 -73.153278   335   
192   BHB          Hancock County - Bar Harbor  44.449769 -68.361565    83   
1240  SLK          Adirondack Regional Airport  44.385300 -74.206200  1663   
157   AUG                        Augusta State  44.320647 -69.797317   352   
799   LKP                  Lake Placid Airport  44.264444 -73.961944  1747   
910   MPV                 Edward F Knapp State  44.203503 -72.562328  1165   
856   ME5                        Banks Airport  44.165389 -68.428167   100   
1155  RKD         Knox County Regional Airport  44.060111 -69.099230    56   
781   LEW                       Lewiston Maine  44.048473 -70.283507   288   

      tz dst             tzone  
1398  -5   A  America/New_York  
259   -5   A  America/New_York  
1101  -5   A  America/New_York  
624   -5   A  America/New_York  
894   -5   A  America/New_York  
923   -5   A  America/New_York  
444   -5   A  America/New_York  
191   -5   A  America/New_York  
1053  -5   A  America/New_York  
933   -5   A  America/New_York  
236   -5   A  America/New_York  
192   -5   A  America/New_York  
1240  -5   A  America/New_York  
157   -5   A  America/New_York  
799   -5   A  America/New_York  
910   -5   A  America/New_York  
856   -5   A  America/New_York  
1155  -5   A  America/New_York  
781   -5   A  America/New_York

### According to the data Northern Aroostook Regional Airport is the easternmost airport in the United States. Double-checking this fact Northern Aroostook Regional Airport latitude is 47 and its longitude in -68, matching the exported data set. However, as of 2022 Lubec Municipal Airport is the easternmost airport with a latitude of 44 and longitude of -67. As mentioned on the town of Lubec, Maine website: https://townoflubec.com/airport/#:~:text=Lubec%20Municipal%20Airport%20is%20the,turf%20which%20is%20maintained%20seasonally.

3. On February 12th, 2013, which New York area airport had the windiest weather?

In [8]:
speed = pd.read_csv("https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/weather.csv")
speed[(speed["month"]==2) & (speed["day"]==12) & (speed["year"]==2013)].sort_values("wind_speed", ascending=False)

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1009     EWR  2013      2   12     3  39.02  26.96  61.63     260.0   
18417    LGA  2013      2   12     2  42.98  26.06  50.94     290.0   
1018     EWR  2013      2   12    12  44.06  26.06  48.87     270.0   
18428    LGA  2013      2   12    13  44.06  23.00  43.02     300.0   
18429    LGA  2013      2   12    14  44.06  23.00  43.02     300.0   
...      ...   ...    ...  ...   ...    ...    ...    ...       ...   
1006     EWR  2013      2   12     0  39.92  39.02  96.55     240.0   
1027     EWR  2013      2   12    21  35.96  24.08  61.67     300.0   
18415    LGA  2013      2   12     0  42.08  37.40  86.89       NaN   
9709     JFK  2013      2   12     0  42.08  37.04  82.17     280.0   
1029     EWR  2013      2   12    23  35.96  23.00  58.94     250.0   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
1009   1048.36058        NaN     0.0    1008.3   10.0  2013-02-12T08:00:00Z  
18417    23.01560   31.07106     0.0    1007.1   10.0  2013-02-12T07:00:00Z  
1018     21.86482   31.07106     0.0    1012.5   10.0  2013-02-12T17:00:00Z  
18428    21.86482   25.31716     0.0    1011.7   10.0  2013-02-12T18:00:00Z  
18429    20.71404   25.31716     0.0    1011.5   10.0  2013-02-12T19:00:00Z  
...           ...        ...     ...       ...    ...                   ...  
1006      6.90468        NaN     0.0    1006.9   10.0  2013-02-12T05:00:00Z  
1027      5.75390        NaN     0.0    1013.0   10.0  2013-02-13T02:00:00Z  
18415     4.60312        NaN     0.0       NaN   10.0  2013-02-12T05:00:00Z  
9709      4.60312        NaN     0.0    1007.1   10.0  2013-02-12T05:00:00Z  
1029      4.60312        NaN     0.0    1013.7   10.0  2013-02-13T04:00:00Z  

[72 rows x 15 columns]

### According to the data EWR (Newark Liberty Internation) has the windiest weather with wind speeds reaching 1048.36058. However, this number is not plausible. This would make LGA (LaGuardia Airport) the airport with the most plausible windiest weather with winds reaching 23. 